### PR TITLE
Nk survey question

### PR DESCRIPTION
**Pull request by NoorAlkattan**
report survey which took by customer with all questions and answer , refactor it without model.

### CODE REVIEW

File: app/models/survey_form.rb [hunk 1]
```diff
@@ -2,7 +2,18 @@ class SurveyForm < ApplicationRecord
   belongs_to :surveyable, polymorphic: true
   has_many :questions, dependent: :destroy
   has_many :answers, through: :questions, dependent: :destroy
-  has_one :category
+  belongs_to :ad, optional: true
+  belongs_to :category, optional: true
```
Comment (kandelchy) [inline app/models/survey_form.rb]: Survey form can not belong to category, each survey has only one category

File: db/schema.rb [hunk 2]
```diff
@@ -72,9 +72,11 @@
     t.bigint "survey_form_id"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.integer "customer_survey_id"
```
Comment (kandelchy) [inline db/schema.rb]: You are not suppose to change the schema, by adding customer_survey_id we have to change the attempts controller.

File: db/schema.rb [hunk 3]
```diff
@@ -336,11 +347,23 @@
     t.integer "sub_type"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.integer "customer_id"
```
Comment (kandelchy) [inline db/schema.rb]: Again you can't change the schema and add fields everywhere

File: db/schema.rb (hunk 2 above)
Comment (NoorAlkattan) [reply db/schema.rb]: done

File: db/schema.rb (hunk 3 above)
Comment (NoorAlkattan) [reply db/schema.rb]: done

File: app/assets/javascripts/customer.js [hunk 4]
```diff
@@ -4,12 +4,39 @@ $(document).on('turbolinks:load', function() {
   language: window.datatableLanguage ||  {}
```
Comment (ssharba) [inline app/assets/javascripts/customer.js]: let's call this file customers and remove .coffee if not needed

File: app/controllers/public_controller.rb [hunk 5]
```diff
@@ -20,5 +21,15 @@ def message_params
         :topic,
       ])
     end
-
+  
+  def prepare_request_data
```
Comment (ssharba) [inline app/controllers/public_controller.rb]: @NoorAlkattan what is this?

Comment (NoorAlkattan) [reply app/controllers/public_controller.rb]: this for feedback datacapture

Comment (ssharba) [reply app/controllers/public_controller.rb]: this not required this week, also you already have PR for that.

Comment (NoorAlkattan) [reply app/controllers/public_controller.rb]: ok i will removed it

File: app/assets/javascripts/customer.js (hunk 4 above)
Comment (NoorAlkattan) [reply app/assets/javascripts/customer.js]: .coffee we need it because it has customer datatable

**Pull request merged**